backend/app/model/mixins/representative_file_version.rb

```diff
@@ -49,17 +65,44 @@ def sequel_to_jsonmodel(objs, opts = {})
           end
         end
 
-        # this could get very ugly
+        if json["representative_file_version"].nil? && self.name == "DigitalObject"
+          file_version_set = DigitalObject
+                               .inner_join(:digital_object_component, root_record_id: :digital_object__id)
+                               .inner_join(:file_version, digital_object_component_id: :digital_object_component__id)
+                               .filter(root_record_id: json.id)
+                               .filter(file_version__publish: true)
+                               .filter(:file_version__id)
+                               .select(Sequel.as(:file_version__id, :file_version_id),
+                                       Sequel.as(:file_version__is_representative, :file_version_is_representative))
+          unless (file_version_set.empty?)
+            file_version_id = file_version_set
+                                .order(:file_version_is_representative)
+                                .reverse
+                                .first[:file_version_id]
+            json["representative_file_version"] = FileVersion.to_jsonmodel(file_version_id)
+          end
+
+        end
+
         if json["representative_file_version"].nil? && self.name == "Resource"
```

Comment:
Minor: same as above, for consistency (but feel free to ignore).

Reply:
🚬